crates/aptos/src/move_tool/bytecode.rs

```diff
@@ -80,6 +84,11 @@ pub struct BytecodeCommand {
 
     #[clap(flatten)]
     pub(crate) prompt_options: PromptOptions,
+
+    /// When `--bytecode-path` is set with this option,
+    /// print out the version of compiler that generated the bytecode
+    #[clap(long)]
+    pub print_compiler_version_only: bool,
```

Comment:
Do we instead want to have a flag for printing out all 3 of:
- compiler version
- language version
- bytecode version
?

Reply:
Perhaps it would be better:

`--print-metadata`

And return all of those in a JSON output, which allows you to add more in the future with backwards compatibility, and keep it all nice and tidy for learning metadata about the code

Reply:
done

Reply:
I think the output is still not fully JSON, which is what @gregnazario may have been asking for. It currently is 3 key value pairs, each printed by itself, which is not quite JSON.

Reply:
yeah, each field is JSON but they are separated. Is there a convenient way to pretty print three separate data structure in a JSON format?

Reply:
Okay to be more prescriptive:

Instead of using `println!` return the object with the data.

The CLI has automatic JSON conversion at the end for the type as long as it is serializable.

Reply:
@gregnazario, sorry for the noob question. The method  `async fn execute(self, command_type: BytecodeCommandType) -> CliTypedResult<String>` requires returning a string. Should I translate the object to string first?

Reply:
If you change the `CliTypedResult<String>` to `CliTypedResult<Struct>`, the outer layer of the CLI will convert it.

Take a look at some of the non-move commands, like the account commands, and you'll be able to see an example

Reply:
Unfortunately we cannot change the return type because we only want to print the metadata when we specify the option `--print-metadata-only` and the main flow of this command needs to return a string with other information.

Reply:
Updated the PR so that it pretty-prints the JSON string as shown in the PR description.